zcash_client_sqlite/src/init.rs

```diff
@@ -0,0 +1,303 @@
+//! Functions for initializing the various databases.
+
+use rusqlite::{types::ToSql, Connection, NO_PARAMS};
+use std::path::Path;
+use zcash_client_backend::encoding::encode_extended_full_viewing_key;
+use zcash_primitives::{block::BlockHash, zip32::ExtendedFullViewingKey};
+
+use crate::{
+    address_from_extfvk,
+    error::{Error, ErrorKind},
+    HRP_SAPLING_EXTENDED_FULL_VIEWING_KEY,
+};
+
+/// Sets up the internal structure of the cache database.
+///
+/// # Examples
+///
+/// ```
+/// use tempfile::NamedTempFile;
+/// use zcash_client_sqlite::init::init_cache_database;
+///
+/// let data_file = NamedTempFile::new().unwrap();
+/// let db_cache = data_file.path();
+/// init_cache_database(&db_cache).unwrap();
+/// ```
+pub fn init_cache_database<P: AsRef<Path>>(db_cache: P) -> Result<(), Error> {
+    let cache = Connection::open(db_cache)?;
+    cache.execute(
+        "CREATE TABLE IF NOT EXISTS compactblocks (
+            height INTEGER PRIMARY KEY,
+            data BLOB NOT NULL
+        )",
+        NO_PARAMS,
+    )?;
+    Ok(())
+}
+
+/// Sets up the internal structure of the data database.
+///
+/// # Examples
+///
+/// ```
+/// use tempfile::NamedTempFile;
+/// use zcash_client_sqlite::init::init_data_database;
+///
+/// let data_file = NamedTempFile::new().unwrap();
+/// let db_data = data_file.path();
+/// init_data_database(&db_data).unwrap();
+/// ```
+pub fn init_data_database<P: AsRef<Path>>(db_data: P) -> Result<(), Error> {
```

Comment:
This should require specifying privacy-relevant policy such as whether `ovk`s are stored.

Reply:
Alternatively (or in addition), `transact::create_to_address` could require specifying a per-transaction policy; this is where the policy would be enforced.

Reply:
Specifically, here is where the current default policy is implemented: https://github.com/zcash/librustzcash/pull/246/files#diff-0db70545aba62e450ec1e1634b293f05R104